src/libsyntax/diagnostics/plugin.rs

```diff
@@ -111,7 +111,7 @@ pub fn expand_register_diagnostic<'cx>(ecx: &'cx mut ExtCtxt,
     // overflow the maximum line width.
     description.map(|raw_msg| {
         let msg = raw_msg.as_str();
-        if !msg.starts_with("\n") || !msg.ends_with("\n") {
+        if !msg.starts_with('\n') || !msg.ends_with('\n') {
```

Comment:
Well, in theory string search effectively works on byte arrays so `"X"` is already what it wants, and `'X'` need is a unicode character and needs to be converted into bytes first, and why would you want to deal with unicode characters at all if you just want to search for a substring.
Something like that.